packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -0,0 +1,162 @@
+
+const cx = classNames.bind(styles);
+
+const Variants = {
+  NEUTRAL: 'neutral',
+  GHOST: 'ghost',
+};
+
+const propTypes = {
+  /**
+   * The options to display in the dropdown. Should be of type `Button`.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Determines whether the primary button and expanding the dropdown should be disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Determines whether the component should have block styles applied. The dropdown will match the component's width.
+   */
+  isBlock: PropTypes.bool,
+  /**
+   * Sets the text that will be shown on the primary button which is outside the dropdown.
+   */
+  primaryOptionLabel: PropTypes.string.isRequired,
+  /**
+   * What will be called when the primary button is pressed.
+   */
+  onClick: PropTypes.func.isRequired,
+  /**
+   * Sets the styles of the component. Must be either 'neutral' or 'ghost'.
+   */
+  variant: PropTypes.oneOf(Object.values(Variants)),
+};
+
+const defaultProps = {
+  disabled: false,
+  isBlock: false,
+  variant: 'neutral',
+};
+
+class SplitButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleDropdownButtonClick = this.handleDropdownButtonClick.bind(this);
+    this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
+    this.handlePrimaryKeyDown = this.handlePrimaryKeyDown.bind(this);
+    this.handlePrimaryKeyUp = this.handlePrimaryKeyUp.bind(this);
+    this.handleChevronKeyDown = this.handleChevronKeyDown.bind(this);
+    this.handleChevronKeyUp = this.handleChevronKeyUp.bind(this);
+
+    this.state = { isOpen: false, chevronActive: false, primaryActive: false };
+  }
+
+  handleDropdownButtonClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleDropdownRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  /*
+    In FireFox active styles don't get applied onKeyDown
+   */
+  handlePrimaryKeyDown(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode == KeyCode.KEY_RETURN) {
+      this.setState({ primaryActive: true });
+    }
+  }
+
+  handlePrimaryKeyUp(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode == KeyCode.KEY_RETURN) {
+      this.setState({ primaryActive: false });
+    }
+  }
+
+  handleChevronKeyDown(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ chevronActive: true });
+    }
+  }
+
+  handleChevronKeyUp(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE) {
+      this.setState({ chevronActive: false });
+    }
+  }
+
+  render() {
+    const {
+      children,
+      disabled,
+      isBlock,
+      primaryOptionLabel,
+      onClick,
+      variant,
+      ...customProps
+    } = this.props;
+
+    const {
+      isOpen,
+      primaryActive,
+      chevronActive
+    } = this.state;
+
+    const finalVariant = variant !== 'emphasis' ? variant : 'neutral';
+
+    return (
+      <DropdownButtonBase
+        {...customProps}
```

Comment:
Do we want to allow customProp spreading?

Reply:
Probably? Most components do.